In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler
from sklearn.pipeline import Pipeline
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import (
    accuracy_score, precision_recall_fscore_support, classification_report,
    confusion_matrix, roc_auc_score, RocCurveDisplay, PrecisionRecallDisplay
)

import warnings
warnings.filterwarnings("ignore")

plt.rcParams["figure.figsize"] = (7,4)
plt.style.use("seaborn-v0_8-whitegrid")



In [3]:
from google.colab import files
uploaded = files.upload()


Saving Ionosphere.csv to Ionosphere.csv


In [4]:
df = pd.read_csv("Ionosphere.csv")
print(df.shape)
df.head()

(351, 35)


Attribute1  Attribute2  Attribute3  Attribute4  Attribute5  Attribute6  \
0           1           0     0.99539    -0.05889     0.85243     0.02306   
1           1           0     1.00000    -0.18829     0.93035    -0.36156   
2           1           0     1.00000    -0.03365     1.00000     0.00485   
3           1           0     1.00000    -0.45161     1.00000     1.00000   
4           1           0     1.00000    -0.02401     0.94140     0.06531   

   Attribute7  Attribute8  Attribute9  Attribute10  ...  Attribute26  \
0     0.83398    -0.37708     1.00000      0.03760  ...     -0.51171   
1    -0.10868    -0.93597     1.00000     -0.04549  ...     -0.26569   
2     1.00000    -0.12062     0.88965      0.01198  ...     -0.40220   
3     0.71216    -1.00000     0.00000      0.00000  ...      0.90695   
4     0.92106    -0.23255     0.77152     -0.16399  ...     -0.65158   

   Attribute27  Attribute28  Attribute29  Attribute30  Attribute31  \
0      0.41078     -0.46168      0.21266     -0.34090      0.42267   
1     -0.20468     -0.18401     -0.19040     -0.11593     -0.16626   
2      0.58984     -0.22145      0.43100     -0.17365      0.60436   
3      0.51613      1.00000      1.00000     -0.20099      0.25682   
4      0.13290     -0.53206      0.02431     -0.62197     -0.05707   

   Attribute32  Attribute33  Attribute34  Target  
0     -0.54487      0.18641     -0.45300       1  
1     -0.06288     -0.13738     -0.02447       0  
2     -0.24180      0.56045     -0.38238       1  
3      1.00000     -0.32382      1.00000       0  
4     -0.59573     -0.04608     -0.65697       1  

[5 rows x 35 columns]

In [10]:
if df.iloc[:,-1].dtype == 'object':
    df.iloc[:,-1] = df.iloc[:,-1].map({'g':1, 'b':0})

X = df.iloc[:, :-1].copy()
y = df.iloc[:,  -1].astype(int).values

X_train, X_test, y_train, y_test = train_test_split(
    X, y, test_size=0.20, random_state=42, stratify=y
)

print("Train:", X_train.shape, " Test:", X_test.shape)


Train: (280, 34)  Test: (71, 34)


In [11]:
logreg = Pipeline([
    ("scaler", StandardScaler()),
    ("clf", LogisticRegression(max_iter=1000, random_state=42))
])

logreg.fit(X_train, y_train)
y_pred = logreg.predict(X_test)

acc = accuracy_score(y_test, y_pred)
print(f"Accuracy: {acc:.3f}")


Accuracy: 0.930


We trained a Logistic Regression model on the Ionosphere dataset.
Features were standardized inside a pipeline.
On the test split, accuracy was 0.930